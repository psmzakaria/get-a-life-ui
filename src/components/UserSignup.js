import React from "react";
import { Button, Checkbox, Form, Divider, Message } from "semantic-ui-react";

const UserSignup = props => {
  return (
    <div className="ui inverted segment">
      <div className="ui inverted form">
        <h1>Sign Up</h1>
        <Message
          negative
          hidden={props.state.hideError}
          header={props.state.errorHeader}
          content={props.state.errorContent}
        />
        <Form>
          <Form.Field>
            <label>Username</label>
            <input
              id="username"
              placeholder="Username"
              value={props.state.username}
              onChange={props.handleOnChange}
            />
          </Form.Field>
          <Form.Field>
            <label>Password</label>
            <input
              id="password"
              placeholder="Password"
              value={props.state.password}
              onChange={props.handleOnChange}
              type="password"
            />
          </Form.Field>
          <Form.Field>
            <Checkbox label="I agree to the Terms and Conditions" />
          </Form.Field>
          <Button fluid color="blue" type="submit" onClick={props.handleSubmit}>
            Submit
          </Button>
        </Form>
        <Divider />
        <div>
          <p className="text-align-center">
            Have an account?
            <a
              className="cursor-pointer"
              onClick={() => props.loadComponent("Signin")}
            >
              {` `} Sign In Here
            </a>
          </p>
        </div>
      </div>
    </div>
  );
};

export default UserSignup;
