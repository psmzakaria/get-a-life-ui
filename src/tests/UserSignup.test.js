import React from "react";
import UserSignup from "./../components/UserSignup";

import ShallowRenderer from "react-test-renderer/shallow";

it("renders the sign up page ", () => {
  const state = {
    hideError: true,
    errorHeader: "",
    errorContent: "",
    username: "",
    password: ""
  };

  const renderer = new ShallowRenderer();
  renderer.render(<UserSignup state={state} />);
  const output = renderer.getRenderOutput();
  expect(output).toMatchSnapshot();
});
