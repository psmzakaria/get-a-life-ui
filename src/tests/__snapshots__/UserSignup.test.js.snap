// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the sign up page  1`] = `
<div
  className="ui inverted segment"
>
  <div
    className="ui inverted form"
  >
    <h1>
      Sign Up
    </h1>
    <Message
      content=""
      header=""
      hidden={true}
      negative={true}
    />
    <Form
      as="form"
    >
      <FormField>
        <label>
          Username
        </label>
        <input
          id="username"
          onChange={undefined}
          placeholder="Username"
          value=""
        />
      </FormField>
      <FormField>
        <label>
          Password
        </label>
        <input
          id="password"
          onChange={undefined}
          placeholder="Password"
          type="password"
          value=""
        />
      </FormField>
      <FormField>
        <Checkbox
          label="I agree to the Terms and Conditions"
          type="checkbox"
        />
      </FormField>
      <Button
        as="button"
        color="blue"
        fluid={true}
        onClick={undefined}
        role="button"
        type="submit"
      >
        Submit
      </Button>
    </Form>
    <Divider />
    <div>
      <p
        className="text-align-center"
      >
        Have an account?
        <a
          className="cursor-pointer"
          onClick={[Function]}
        >
           
           Sign In Here
        </a>
      </p>
    </div>
  </div>
</div>
`;
